### ISSUE TITLE
Bool vs Literal True/False

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [microsoft/pyright#1797]
**Describe the bug**
The type it chooses is too specific and it gets stuck not seeing that the dict is equivalent to Dict[Tuple[bool, bool], str]

**To Reproduce**
```
RESPONSES = {
    (True, True): "A",
    (True, False): "B",
    (False, True): "C",
    (False, False): "D"
}


def bool_to_response(x: bool, y: bool) -> str:
    return RESPONSES[(x, y)]

```

Error given is, 

Argument of type "tuple[bool, bool]" cannot be assigned to parameter "k" of type "tuple[Literal[True], Literal[True]] | tuple[Literal[True], Literal[False]] | tuple[Literal[False], Literal[True]] | tuple[Literal[False], Literal[False]]" in function "__getitem__"
  Type "tuple[bool, bool]" cannot be assigned to type "tuple[Literal[True], Literal[True]] | tuple[Literal[True], Literal[False]] | tuple[Literal[False], Literal[True]] | tuple[Literal[False], Literal[False]]"
    Tuple entry 1 is incorrect type
      "bool" cannot be assigned to type "Literal[True]"
    Tuple entry 1 is incorrect type
      "bool" cannot be assigned to type "Literal[True]"
    Tuple entry 1 is incorrect type
      "bool" cannot be assigned to type "Literal[False]"
    Tuple entry 1 is incorrect type

**Expected behavior**
No error.

**VS Code extension or command-line**
2021.4.3.-pre.1 pylance

**Additional context**
This is a minor thing and I can fix it with an explicit type declaration if it's messy to fix or just too much work for the type checker as I'm guessing other similar cases of triple bools also get stuck.

**Comment by erictraut** [comment]
This is the intended behavior. For details about pyright's type inference behavior for tuples, refer to [this documentation](https://github.com/microsoft/pyright/blob/master/docs/type-inference.md#tuple-expressions). This behavior has been refined over time to handle most cases most of the time, but there will be times when you need to override the inference rules with an explicit type annotation. This is one of those cases.

```python
RESPONSES: Dict[Tuple[bool, bool], str] = {
}
```

**Comment by erictraut** [comment]
After much deliberation, I've decided to revisit this issue and change the way that pyright infers tuple expressions. In particular, when a tuple expression is embedded within another container (list, dict, set, or tuple) expression, the type checker will no longer retain literal values for the tuple. This change will be included in the next release of pyright.

**Comment by erictraut** [comment]
This is included in pyright 1.1.365.
